src/improvements/tasks/OPCMBaseTask.sol

```diff
@@ -73,7 +69,17 @@ abstract contract OPCMBaseTask is L2TaskBase {
             _stateInfos[parentMultisig].length == 1,
             "OPCMBaseTask: only nonce should be updated on upgrade controller multisig"
         );
-        require(_stateInfos[opcm()].length == 0, "OPCMBaseTask: Storage writes are not allowed on OPCM");
+
+        bytes32 opcmStateSlot = bytes32(uint256(stdstore.target(opcm()).sig(IOPContractsManager.isRC.selector).find()));
+        require(_stateInfos[opcm()].length <= 1, "OPCMBaseTask: OPCM must have at most 1 state change");
+        // Not all invocations of OPCM upgrade will have the isRC state change. This is because it only happens when
+        // address(this) is equal to the OPCMs 'upgradeController' address (which is an immutable).
+        if (_stateInfos[opcm()].length == 1) {
+            StateInfo storage stateInfo = _stateInfos[opcm()][0];
+            require(stateInfo.slot == opcmStateSlot, "OPCMBaseTask: Incorrect OPCM isRc slot");
+            require(stateInfo.oldValue == bytes32(uint256(1)), "OPCMBaseTask: Incorrect OPCM isRc old value");
+            require(stateInfo.newValue == bytes32(uint256(0)), "OPCMBaseTask: Incorrect OPCM isRc new value");
+        }
```

Comment:
same AccountAccessParser comment

Reply:
Addressed in: https://github.com/ethereum-optimism/superchain-ops/pull/664/commits/c908437b9e807b01b16c9867d5f9b15aa62d615b